### DISCUSSION TITLE
Use sweeper result in consecutive training run

### DISCUSSION THREAD

**Discussion by faroit** [facebookresearch/hydra#2380] [Q&A]
We are using a sweeper plugin to find a learning rate for a lightning system. 

What is the best workflow to use the returned optimized hyperparameters from the sweeper in a new run?

### Example

the sweeper is configures with

```yaml
hydra:
  sweeper:
    _target_: hydra_plugins.hydra_optuna_sweeper.optuna_sweeper.OptunaSweeper
    params:
      optimizer.lr: interval(0.00001, 0.01)
```

```
python train.py hydra/sweeper=optuna
python train.py optimizer.lr=$optimized_value$
```

I guess there are a couple of options:

- read the sweeper result file and extract the parameters from there manually e.g. `python train.py sweeper_result_path=...` Likely related to https://github.com/facebookresearch/hydra/issues/468
- grab the optimized values from the logger when the sweeper has finished and convert them to variables using bash to set the optimized value
- modify the sweeper or add a callback to set environment variables with the optimized values
- ...

Did I miss something? Which one would you recommend?
